File: src/Entity/Activite.php
<?php

namespace App\Entity;

use App\Repository\ActiviteRepository;
use Doctrine\ORM\Mapping as ORM;
use App\Entity\Partenaire;

#[ORM\Entity(repositoryClass: ActiviteRepository::class)]
#[ORM\Table(name: 'activite')]
class Activite
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(nullable: true, length: 255)]
    private ?string $nom = null;

    #[ORM\Column(length: 80)]
    private string $type = '';

    #[ORM\Column]
    private string $genreCible = '';

    #[ORM\Column(length: 120)]
    private string $lieu = '';

    #[ORM\Column(type: 'datetime')]
    private ?\DateTimeInterface $dateActivite = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    #[ORM\Column(type: 'time', nullable: true)]
    private ?\DateTimeInterface $createdAt = null;

    #[ORM\ManyToOne(targetEntity: Partenaire::class)]
    #[ORM\JoinColumn(nullable: true)]
    private ?Partenaire $partenaire = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $images = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private string $prix = '0.00';

    #[ORM\Column(type: 'integer')]
    private int $nombrePlaces = 0;

    #[ORM\Column(type: 'integer')]
    private int $placesDisponibles = 0;

    #[ORM\Column(type: 'date', nullable: true)]
    private ?\DateTimeInterface $dateCreation = null;

    #[ORM\Column(type: 'boolean')]
    private bool $avecDate = false;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(?string $nom): static
    {
        $this->nom = $nom;
        return $this;
    }

    public function getType(): string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        $this->type = $type;
        return $this;
    }

    public function getGenreCible(): string
    {
        return $this->genreCible;
    }

    public function setGenreCible(string $genreCible): static
    {
        $this->genreCible = $genreCible;
        return $this;
    }

    public function getLieu(): string
    {
        return $this->lieu;
    }

    public function setLieu(string $lieu): static
    {
        $this->lieu = $lieu;
        return $this;
    }

    public function getDateActivite(): ?\DateTimeInterface
    {
        return $this->dateActivite;
    }

    public function setDateActivite(?\DateTimeInterface $dateActivite): static
    {
        $this->dateActivite = $dateActivite;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): static
    {
        $this->description = $description;
        return $this;
    }

    public function getCreatedAt(): ?\DateTimeInterface
    {
        return $this->createdAt;
    }

    public function setCreatedAt(?\DateTimeInterface $createdAt): static
    {
        $this->createdAt = $createdAt;
        return $this;
    }

    public function getPartenaire(): ?Partenaire
    {
        return $this->partenaire;
    }

    public function setPartenaire(?Partenaire $partenaire): static
    {
        $this->partenaire = $partenaire;
        return $this;
    }

    public function getImages(): ?string
    {
        return $this->images;
    }

    public function setImages(?string $images): static
    {
        $this->images = $images;
        return $this;
    }

    public function getPrix(): string
    {
        return $this->prix;
    }

    public function setPrix(string $prix): static
    {
        $this->prix = $prix;
        return $this;
    }

    public function getNombrePlaces(): int
    {
        return $this->nombrePlaces;
    }

    public function setNombrePlaces(int $nombrePlaces): static
    {
        $this->nombrePlaces = $nombrePlaces;
        return $this;
    }

    public function getPlacesDisponibles(): int
    {
        return $this->placesDisponibles;
    }

    public function setPlacesDisponibles(int $placesDisponibles): static
    {
        $this->placesDisponibles = $placesDisponibles;
        return $this;
    }

    public function getDateCreation(): ?\DateTimeInterface
    {
        return $this->dateCreation;
    }

    public function setDateCreation(?\DateTimeInterface $dateCreation): static
    {
        $this->dateCreation = $dateCreation;
        return $this;
    }

    public function getAvecDate(): bool
    {
        return $this->avecDate;
    }

    public function setAvecDate(bool $avecDate): static
    {
        $this->avecDate = $avecDate;
        return $this;
    }

}

File: nexora_java/src/main/java/com/pi/entities/Activite.java
package com.pi.entities;

import java.time.LocalDate;

public class Activite {
    private int id;
    private String nom;
    private String description;
    private String type;
    private String lieu;
    private LocalDate dateActivite;
    private String images;
    private double prix;
    private int nombrePlaces;
    private int placesDisponibles;
    private int partenaireId;
    private LocalDate dateCreation;
    private String partenaireNom;
    private String partenaireEmail;
    private String partenaireTelephone;

    public Activite() {
        this.dateCreation = LocalDate.now();
    }

    public Activite(String nom, String type, String lieu, LocalDate dateActivite,
                    String images, double prix, int nombrePlaces, int partenaireId) {
        this.nom = nom;
        this.type = type;
        this.lieu = lieu;
        this.dateActivite = dateActivite;
        this.images = images;
        this.prix = prix;
        this.nombrePlaces = nombrePlaces;
        this.placesDisponibles = nombrePlaces;
        this.partenaireId = partenaireId;
        this.dateCreation = LocalDate.now();
    }

    public Activite(String nom, String description, String type, String lieu, LocalDate dateActivite,
                    String images, double prix, int nombrePlaces, int partenaireId) {
        this.nom = nom;
        this.description = description;
        this.type = type;
        this.lieu = lieu;
        this.dateActivite = dateActivite;
        this.images = images;
        this.prix = prix;
        this.nombrePlaces = nombrePlaces;
        this.placesDisponibles = nombrePlaces;
        this.partenaireId = partenaireId;
        this.dateCreation = LocalDate.now();
    }

    // Getters et Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getLieu() {
        return lieu;
    }

    public void setLieu(String lieu) {
        this.lieu = lieu;
    }

    public LocalDate getDateActivite() {
        return dateActivite;
    }

    public void setDateActivite(LocalDate dateActivite) {
        this.dateActivite = dateActivite;
    }

    public String getImages() {
        return images;
    }

    public void setImages(String images) {
        this.images = images;
    }

    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public int getNombrePlaces() {
        return nombrePlaces;
    }

    public void setNombrePlaces(int nombrePlaces) {
        this.nombrePlaces = nombrePlaces;
    }

    public int getPlacesDisponibles() {
        return placesDisponibles;
    }

    public void setPlacesDisponibles(int placesDisponibles) {
        this.placesDisponibles = placesDisponibles;
    }

    public int getPartenaireId() {
        return partenaireId;
    }

    public void setPartenaireId(int partenaireId) {
        this.partenaireId = partenaireId;
    }

    public LocalDate getDateCreation() {
        return dateCreation;
    }

    public void setDateCreation(LocalDate dateCreation) {
        this.dateCreation = dateCreation;
    }

    public String getPartenaireNom() {
        return partenaireNom;
    }

    public void setPartenaireNom(String partenaireNom) {
        this.partenaireNom = partenaireNom;
    }

    public String getPartenaireEmail() {
        return partenaireEmail;
    }

    public void setPartenaireEmail(String partenaireEmail) {
        this.partenaireEmail = partenaireEmail;
    }

    public String getPartenaireTelephone() {
        return partenaireTelephone;
    }

    public void setPartenaireTelephone(String partenaireTelephone) {
        this.partenaireTelephone = partenaireTelephone;
    }

    @Override
    public String toString() {
        return "Activite{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", description='" + description + '\'' +
                ", type='" + type + '\'' +
                ", lieu='" + lieu + '\'' +
                ", dateActivite=" + dateActivite +
                ", prix=" + prix +
                ", placesDisponibles=" + placesDisponibles +
                '}';
    }
}
